components/nimble/sources/new-contact-created/new-contact-created.mjs

```diff
@@ -0,0 +1,68 @@
+import app from "../../nimble.app.mjs";
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+
+export default {
+  name: "New Contact Created",
+  version: "0.0.1",
+  key: "nimble-new-contact-created",
+  description: "Emit new event when a contact is created.",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    app,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      static: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+  },
+  methods: {
+    emitEvent(data) {
+      this.$emit(data, {
+        id: `${data.id}`,
+        summary: `New contact created with ID ${data.id}`,
+        ts: Date.parse(data.created),
+      });
+    },
+    _setLastResourceId(id) {
+      this.db.set("lastResourceId", id);
+    },
+    _getLastResourceId() {
+      return this.db.get("lastResourceId");
+    },
+  },
+  async run() {
+    const { resources } = await this.app.getContacts();
+
+    resources.filter((contact) => contact.record_type === "person")
+      .forEach(this.emitEvent);
+
+    const lastResourceId = this._getLastResourceId();
+
+    let page = 1;
+
+    while (page >= 0) {
+      let { resources } = await this.webinargeek.getContacts({
+        params: {
+          page,
+          per_page: 100,
+        },
+      });
+
+      resources = resources.filter((contact) => contact.record_type === "person");
+
+      resources.reverse().forEach(this.emitEvent);
+
+      if (
+        resources.length < 100 ||
+        resources.filter((resource) => resource.id === lastResourceId)
```

Comment:
This will always be truthy, since `.filter` will always return an array (even if empty)
I think your intention might've been to use `.find` instead